### PR TITLE
GetAccountAssets: pagination of response

### PR DESCRIPTION
**Pull request by MBoldyrev**
Signed-off-by: Mikhail Boldyrev <miboldyrev@gmail.com>

[IR-473](https://jira.hyperledger.org/browse/IR-473)
[POW-090](https://soramitsu.atlassian.net/wiki/spaces/IS/pages/1120239703/POW+N+090+AccountAsset+Pagination)

### Description of the Change

Page metadata added to GetAccountAssets query response.

### Benefits

One step towards pagination of GetAccountAssets query.

### CODE REVIEW

File: shared_model/schema/qry_responses.proto [hunk 1]
```diff
@@ -37,6 +37,10 @@ message AccountAsset {
 // *** Responses *** //
 message AccountAssetResponse {
   repeated AccountAsset account_assets = 1;
+  uint32 total_number = 2;
+  oneof opt_next_asset_id {
```
Comment (luckychess) [inline shared_model/schema/qry_responses.proto]: Why we don't use `optional ` modifier?

File: shared_model/backend/protobuf/query_responses/impl/proto_account_asset_response.cpp [hunk 2]
```diff
@@ -14,7 +14,15 @@ namespace shared_model {
         : CopyableProto(std::forward<QueryResponseType>(queryResponse)),
           account_asset_response_{proto_->account_assets_response()},
           account_assets_{account_asset_response_.account_assets().begin(),
-                          account_asset_response_.account_assets().end()} {}
+                          account_asset_response_.account_assets().end()},
+          next_asset_id_{[this]() -> decltype(next_asset_id_) {
+            switch (account_asset_response_.opt_next_asset_id_case()) {
```
Comment (luckychess) [inline shared_model/backend/protobuf/query_responses/impl/proto_account_asset_response.cpp]: Maybe just `if..else` or even `?..:` ?

File: shared_model/schema/qry_responses.proto (hunk 1 above)
Comment (MBoldyrev) [reply shared_model/schema/qry_responses.proto]: AFAIK, `optional` modifier is a [proto2](https://developers.google.com/protocol-buffers/docs/proto#optional) feature, while we use [proto3](https://developers.google.com/protocol-buffers/docs/proto3) that does not have it.

**Comment by lebdron** [comment]
Done in https://github.com/hyperledger/iroha/pull/98

**Pull request closed without merging**